File: vulnerable.ts
import * as http from 'http';

// 1. Secret en dur (Sera détecté par p/secrets)
const STRIP_SECRET_KEY = "sk_live_1234567890abcdef1234567890abcdef";
const DB_PASSWORD = "super_secret_admin_password_123!";

function processData(userInput: string) {
    // 2. Utilisation dangereuse de eval() (Sera détecté par p/default)
    const result = eval(userInput);
    console.log(result);
}

// 3. Vulnérabilité XSS (Cross-Site Scripting)
http.createServer((req, res) => {
    const url = new URL(req.url || '', `http://${req.headers.host}`);
    // Récupération d'un paramètre non sécurisé depuis l'URL
    const userName = url.searchParams.get('name') || 'Visiteur';

    res.writeHead(200, { 'Content-Type': 'text/html' });
    // DANGER : Injection directe de l'entrée utilisateur dans le HTML
    res.write(`<h1>Bienvenue, ${userName}</h1>`); 
    res.end();
}).listen(8080);

File: vulnerable.py
import sqlite3
import subprocess
import os

# 1. Secret en dur (Sera détecté par p/secrets)
AWS_SECRET_ACCESS_KEY = "AKIAIOSFODNN7EXAMPLE"
GITHUB_API_TOKEN = "ghp_xYzAbCdEfGhIjKlMnOpQrStUvWxYz123456"

def get_user(username):
    """2. Injection SQL (Sera détecté par p/security-audit)"""
    conn = sqlite3.connect('users.db')
    cursor = conn.cursor()
    
    # DANGER : Concaténation directe de l'entrée utilisateur dans la requête SQL
    query = f"SELECT * FROM users WHERE username = '{username}'"
    cursor.execute(query)
    
    return cursor.fetchall()

def ping_server(ip_address):
    """3. Injection de commande (Sera détecté par p/security-audit)"""
    # DANGER : Exécution d'une commande shell avec une variable non nettoyée
    command = "ping -c 1 " + ip_address
    subprocess.call(command, shell=True)

def calculate_discount(user_formula):
    """4. Utilisation dangereuse de eval() (Sera détecté par p/default)"""
    # DANGER : Évaluation directe de code arbitraire
    return eval(user_formula)
